carbonmark-api/src/routes/users/put.ts

```diff
@@ -11,12 +13,32 @@ const handler = (fastify: FastifyInstance) =>
     const { wallet, username, description, profileImgUrl } = request.body;
 
     try {
+      /** Get the existing doc with latest nonce */
+      const userDoc = await getFirestoreUserDoc({
+        docId: wallet.toUpperCase(),
+        firestore: fastify.firebase.firestore(),
+      });
+      if (!userDoc) {
+        return reply
+          .code(404)
+          .send({ error: "No user profile found for given handle or address" });
+      }
+
+      const isAuthenticated = verifyProfileSignature({
+        nonce: userDoc.nonce,
+        expectedAddress: userDoc.address,
+        signature: request.headers.authorization?.split(" ")[1] || "",
+      });
+      if (!isAuthenticated) {
+        return reply.status(403).send();
+      }
+
       const updatedData = {
-        username,
-        description,
+        username: username || userDoc.username || userDoc.handle,
+        description: description || userDoc.description || null,
```

Comment:
Description is optionalin the UI. 
I think that with this code we discard "" as a value for description, which prevents the users from deleting the description. (I did not test)

Reply:
Great catch, I also caught a mismatch in the schema which was disallowing empty string. I'll see if I can add a testcase for this.